src/Entity/User.php

```diff
@@ -763,6 +763,11 @@ public function removeOAuth2UserConsent(OAuth2UserConsent $oAuth2UserConsent): s
         return $this;
     }
 
+    public function isSsoControlled(): bool
+    {
+        return $this->oauthGithubId || $this->oauthGoogleId || $this->oauthFacebookId || $this->oauthKeycloakId || $this->oauthZitadelId;
```

Comment:
this is obviously prone to getting out of date if new identity providers are added. Not sure if there's a better way, or maybe can think of one later. We could add a column on user that they are sso controlled, but that means we have to be sure to set it to true every time a new provider is added which is also possible to mess up, but perhaps less likely